### PR TITLE
Implement langversion switch for openstaticclasses

### PR DESCRIPTION
**Pull request by KevinRansom**
1.    Add switch support to product
2.    Plumb tc globals a bit further down
3.    neg tests for failing to find open static methods
4.    Enable versioning tests

### CODE REVIEW

File: src/fsharp/TypeChecker.fs [hunk 1]
```diff
@@ -645,7 +645,7 @@ let ImplicitlyOpenOwnNamespace tcSink g amap scopem enclosingNamespacePath env =
         match enclosingNamespacePathToOpen with
         | id :: rest ->
             let ad = env.eAccessRights
-            match ResolveLongIndentAsModuleOrNamespaceOrStaticClass tcSink ResultCollectionSettings.AllResults amap scopem true true OpenQualified env.eNameResEnv ad id rest true with 
+            match ResolveLongIndentAsModuleOrNamespaceOrStaticClass g tcSink ResultCollectionSettings.AllResults amap scopem true true OpenQualified env.eNameResEnv ad id rest true with
```
Comment (TIHan) [inline src/fsharp/TypeChecker.fs]: You don't need to pass `TcGlobals`, `amap: Import.ImportMap` already has `g: TcGlobals` on it. So I would make this function signature to go back to the way it was.

File: src/fsharp/NameResolution.fs [hunk 2]
```diff
@@ -1992,13 +1996,16 @@ let CheckForTypeLegitimacyAndMultipleGenericTypeAmbiguities
 //-------------------------------------------------------------------------
 
 /// Perform name resolution for an identifier which must resolve to be a namespace or module.
-let rec ResolveLongIndentAsModuleOrNamespaceOrStaticClass sink (atMostOne: ResultCollectionSettings) amap m allowStaticClasses first fullyQualified (nenv: NameResolutionEnv) ad (id:Ident) (rest: Ident list) isOpenDecl =
+let rec ResolveLongIndentAsModuleOrNamespaceOrStaticClass (g: TcGlobals) sink (atMostOne: ResultCollectionSettings) amap m allowStaticClasses first fullyQualified (nenv: NameResolutionEnv) ad (id:Ident) (rest: Ident list) isOpenDecl =
+
+    // If the selected language version doesn't support open static classes then turn them off.
+    let allowStaticClasses = allowStaticClasses && g.langVersion.SupportsFeature LanguageFeature.OpenStaticClasses
```
Comment (TIHan) [inline src/fsharp/NameResolution.fs]: This should be: `allowStaticClasses && amap.g.landVersion.SupportsFeature LanguageFeature.OpenStaticClasses` - don't need to pass `TcGlobals` as `amap` already has it.

Comment (TIHan) [reply src/fsharp/NameResolution.fs]: Seems `ImportMap` is not a record and because of that, we need to type it, `amap: Import.ImportMap`

**Comment by dsyme** [comment]
Great! thanks!

**Pull request merged**